### PR TITLE
develop_-_security_fixes

### PR DESCRIPTION
**Pull request by syphzur**

### CODE REVIEW

File: PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/security/custom/CustomBasicAuthenticationFilter.java [hunk 1]
```diff
@@ -59,21 +62,26 @@ private Authentication getLoginPasswordAuthentication(final HttpServletRequest r
...
+            final String login = decodedJwt.getSubject();
+            if (login != null && isNotLoggedOut(tokenWithoutPrefix)) {
+                final Optional<User> foundUser = userRepository.findUserByLogin(login);
+                if (foundUser.isPresent()) {
+                    final UsernamePasswordAuthenticationToken auth = new UsernamePasswordAuthenticationToken(login, null, foundUser
+                            .get()
+                            .getAuthorities());
+                    return auth;
+                }
+                return null;
```
Comment (dsuchy) [inline PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/security/custom/CustomBasicAuthenticationFilter.java]: @syphzur why returm here whem ur return next line anyway

File: PhotoPlusFrontend/PhotoPlus/src/app/services/login/login.service.ts [hunk 2]
```diff
@@ -41,25 +44,26 @@ export class LoginService {
...
+        sessionStorage.setItem("date", date);
     }
 
     public isLoggedIn() {
-        if (localStorage.getItem("token") == null) {
+        const token = sessionStorage.getItem("token")
+        if (token == null) {
             return false
         }
-        //TODO:Check if not expired
```
Comment (dsuchy) [inline PhotoPlusFrontend/PhotoPlus/src/app/services/login/login.service.ts:61]: finally

File: PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/security/custom/CustomBasicAuthenticationFilter.java (hunk 1 above)
Comment (syphzur) [reply PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/security/custom/CustomBasicAuthenticationFilter.java]: because i need to return here

Comment (dsuchy) [reply PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/security/custom/CustomBasicAuthenticationFilter.java]: no

**Review by dsuchy — APPROVED** [review]

**Pull request merged**